tests/test_sof_basic/src/main.xc

```diff
@@ -0,0 +1,100 @@
+// Copyright 2016-2021 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+#include "shared.h"
+
+#define EP_COUNT_OUT   (6)
+#define EP_COUNT_IN    (6)
+
+/* Endpoint type tables */
+XUD_EpType epTypeTableOut[EP_COUNT_OUT] = {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL};
+XUD_EpType epTypeTableIn[EP_COUNT_IN] =   {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL};
+
+void exit(int);
+
+unsigned TestEp_Bulk(chanend c_out, chanend c_in, int epNum, chanend c_sof)
+{
+    unsigned int length;
+    XUD_Result_t res;
+
+    XUD_ep ep_out = XUD_InitEp(c_out);
+    XUD_ep ep_in  = XUD_InitEp(c_in);
+
+    unsigned frames[10];
+
+    /* Buffer for Setup data */
+    unsigned char buffer[1024];
+
+    XUD_GetBuffer(ep_out, buffer, length);
+
+    if(length != 10)
+    {
+        printintln(length);
+        return FAIL_RX_LENERROR;
+    }
+
+    if(RxDataCheck(buffer, length, epNum, 10))
+    {
+        return FAIL_RX_DATAERROR;
+    }
+
+    /* Receive SOFs */
+    for (int i = 0; i< 5; i++) 
+        frames[i] = inuint(c_sof);
+
+    XUD_GetBuffer(ep_out, buffer, length);
+
+    if(length != 11)
+    {
+        printintln(length);
+        return FAIL_RX_LENERROR;
+    }
+
+    if(RxDataCheck(buffer, length, epNum, 11))
+    {
+        return FAIL_RX_DATAERROR;
+    }
+
+    unsigned expectedFrame = 52;
+
+    /* Check frame numbers */
+    for (int i = 0 ; i < 5; i++)
+    {
+        if(frames[i] != i+expectedFrame)
+        {
+            printhexln(i);
+            printhexln(frames[i]);
+            return FAIL_RX_FRAMENUMBER;
+        }
+    }
+
+    return 0;
+}
+
```

Comment:
Quite a few main.xc files `#include "test_main.xc"` and use its common functions.  This file does not.  Is that intentional or an oversight?